core/coreunix/add.go

```diff
@@ -44,6 +45,9 @@ func Add(n *core.IpfsNode, r io.Reader) (string, error) {
 
 // AddR recursively adds files in |path|.
 func AddR(n *core.IpfsNode, root string) (key string, err error) {
+	unlock := n.Blockstore.RLock()
+	defer unlock()
```

Comment:
can we make these `RLock()` calls part of the previous PR? this is what i want to CR carefully-- make sure we call `RLock()` everywhere we need it. (yes that means the previous PR would only introduce `RLock()` calls and no `Lock()`.